### ISSUE TITLE
[Feature Request] Native Jev Mode (/v1/systemone Endpoint) for Fast Typed Decisions & Logit Probabilities

### ISSUE THREAD

**Issue by Psanyi89** [AtomicBot-ai/atomic-llama-cpp-turboquant#84]
## Summary
Request to add native **Jev Mode** (`/v1/systemone` API endpoint) support into `llama-server`. 

Jev Mode replaces free-form text generation with single-pass, typed decisions (Yes/No, single-choice, ordinal scoring) that evaluate normalized log probability distributions over specific candidate tokens directly from a single prefill pass.

Adding native support in `atomic-llama-cpp-turboquant` would pair ultra-fast KV-cache lookup (via TurboQuant) with instant 1-token structured decision evaluation for agent routing loops.

---

## Motivation & Use Case
In agentic architectures (like `atomic-agent`, OpenClaw, or Hermes), a significant fraction of LLM calls are purely for decision-making (e.g., *"Should I search the web? [Yes/No]"*, *"Select tool: [A/B/C/D]"*). 

Generating full text or JSON schemas for these simple branches incurs unnecessary decoding overhead. Jev Mode allows the model to answer typed decision questions in a **single forward pass (0 generated tokens)** by evaluating the log probabilities (`n_probs`) of valid choice labels constrained via grammar.

Combined with **TurboQuant KV cache compression**, this yields sub-10ms decision speeds locally.

---

## Technical Overview: How Jev Mode Works

1. **Typed Questions**: Accepts a batch of structured decision tasks:
   - **Binary**: `[Yes, No]` / `[True, False]`
   - **Single-Choice**: `[Option A, Option B, Option C, ...]`
   - **Ordinal Score**: `[1, 2, 3, 4, 5]`
2. **Constrained Sampling / Masking**: Uses GBNF grammars or logits-masking to enforce candidate tokens at the decision token position.
3. **Logprob Extraction**: Reads the raw softmax logits or `n_probs` at the decision index following prompt prefill.
4. **Probability Distribution**: Normalizes candidate probabilities into a sum-to-1 confidence score (with optional temperature scaling/calibration).

---

### Endpoint
`POST /v1/systemone` (or `POST /v1/jev`)

### Example Request
```json
{
  "messages": [
    { "role": "user", "content": "The user asks: 'What is the weather in Tokyo?'" }
  ],
  "questions": [
    {
      "id": "q1",
      "type": "single_choice",
      "question": "Which tool is required?",
      "options": ["web_search", "calculator", "none"]
    },
    {
      "id": "q2",
      "type": "boolean",
      "question": "Is this a time-sensitive query?",
      "options": ["yes", "no"]
    }
  ]
}

### Example Response
{
  "id": "jev-dec-12345",
  "model": "Qwen3.5-35B-A3B",
  "results": [
    {
      "id": "q1",
      "decision": "web_search",
      "confidence": 0.942,
      "probabilities": {
        "web_search": 0.942,
        "calculator": 0.005,
        "none": 0.053
      }
    },
    {
      "id": "q2",
      "decision": "yes",
      "confidence": 0.887,
      "probabilities": {
        "yes": 0.887,
        "no": 0.113
      }
    }
  ]
}```

Proposed Implementation Guide in llama-server
Server Handler:

Add a route handler for /v1/systemone in tools/server/server.cpp.

Prompt Construction & Caching:

Utilize existing llama_server prompt-cache mechanisms to maintain base message state across multiple decision questions.

Logit Reading:

Compute logits for candidate option tokens at the target position (similar to n_probs implementation in /completion).

Softmax Normalization:

Apply softmax over only the candidate option token IDs to construct the probability distribution.

TurboQuant Integration:

Leverage atomic-llama-cpp-turboquant's WHT-rotated KV-cache quantization to keep prompt state prefill latency as low as possible.

Reference Repositories & Prior Art
Existing open-source implementations and wrappers that can serve as technical reference guides:

NakliTechie/llamacpp-jev

[github.com/NakliTechie/llamacpp-jev](https://github.com/NakliTechie/llamacpp-jev?utm_source=gemini)

Implements /v1/systemone directly over standard llama-server using GBNF grammars, /apply-template, n_probs logprobs, and prompt-cache checkpoints.

nokia-applied-research/AnyJev

[github.com/nokia-applied-research/AnyJev](https://github.com/nokia-applied-research/AnyJev?utm_source=gemini)

Demonstrates single-prefill decision heads, zero-shot logit extraction (L0), and head calibration (L1/L2) over open LLMs.

githubnext/localjev

[github.com/githubnext/localjev](https://github.com/githubnext/localjev?utm_source=gemini)

Bridge protocol mapping TypeSafe Jev system-one decisions to local LLM engines.

Value Add for atomic-llama-cpp-turboquant
Allows local agents running on atomic-llama-cpp-turboquant to eliminate decoding latencies for classification and routing tasks.

Maximizes the throughput gains of TurboQuant KV cache + MTP setup by bypassing multi-token generation when only a categorical decision is needed.
